src/compute-client/src/plan/mod.rs

```diff
@@ -1105,6 +1105,9 @@ impl<T: timely::progress::Timestamp> Plan<T> {
                     assert!(pre_existing.is_none());
                     lir_values.push(lir_value);
                 }
+                for id in ids.iter() {
+                    arrangements.insert(Id::Local(*id), AvailableCollections::new_raw());
+                }
```

Comment:
Can you explain why this is needed?

Reply:
Yes, and will do. But let me try it out here: as we leave the iterative scope, any arrangements we had need to be discarded, as they are arrangements on an inner timestamp. This is the point where we are about to plan `body`, and it does not have access to the inner data in arranged form.